internal/bminventory/inventory.go

```diff
@@ -2195,47 +2206,131 @@ func (b *bareMetalInventory) updateClusterData(_ context.Context, cluster *commo
+			}
+
+			if err = network.VerifyNetworkHostPrefix(clusterNetwork.HostPrefix); err != nil {
+				return common.NewApiError(http.StatusBadRequest, errors.Wrapf(err, "Cluster network host prefix %d", clusterNetwork.HostPrefix))
+			}
+
+			err = network.VerifyClusterCidrSize(int(clusterNetwork.HostPrefix), string(clusterNetwork.Cidr), len(cluster.Hosts))
+			if err != nil {
+				return common.NewApiError(http.StatusBadRequest, errors.Wrap(err, "Cluster CIDR size"))
+			}
 		}
-		clusterCidr = *params.ClusterUpdateParams.ClusterNetworkCidr
-		updates["cluster_network_cidr"] = clusterCidr
+		cluster.ClusterNetworks = params.ClusterUpdateParams.ClusterNetworks
 	}
-	if params.ClusterUpdateParams.ClusterNetworkHostPrefix != nil {
-		if err = network.VerifyNetworkHostPrefix(*params.ClusterUpdateParams.ClusterNetworkHostPrefix); err != nil {
-			return common.NewApiError(http.StatusBadRequest, err)
+
+	if params.ClusterUpdateParams.ServiceNetworks != nil && len(params.ClusterUpdateParams.ServiceNetworks) > 0 {
+		for _, serviceNetwork := range params.ClusterUpdateParams.ServiceNetworks {
+			if err = network.VerifyClusterOrServiceCIDR(string(serviceNetwork.Cidr)); err != nil {
+				return common.NewApiError(http.StatusBadRequest, errors.Wrapf(err, "Service network CIDR %s", string(serviceNetwork.Cidr)))
+			}
 		}
-		hostNetworkPrefix = *params.ClusterUpdateParams.ClusterNetworkHostPrefix
-		updates["cluster_network_host_prefix"] = hostNetworkPrefix
+		cluster.ServiceNetworks = params.ClusterUpdateParams.ServiceNetworks
 	}
-	if clusterCidr != "" {
-		err = network.VerifyClusterCidrSize(int(hostNetworkPrefix), clusterCidr, len(cluster.Hosts))
-		if err != nil {
+
+	if params.ClusterUpdateParams.MachineNetworks != nil && len(params.ClusterUpdateParams.MachineNetworks) > 0 {
+		cluster.MachineNetworks = params.ClusterUpdateParams.MachineNetworks
+	}
+
+	for index := range cluster.MachineNetworks {
+		if string(cluster.MachineNetworks[index].Cidr) != "" {
+			if err = network.VerifyMachineCIDR(string(cluster.MachineNetworks[index].Cidr)); err != nil {
+				return common.NewApiError(http.StatusBadRequest, errors.Wrapf(err, "Machine network CIDR %s", string(cluster.MachineNetworks[index].Cidr)))
+			}
+		}
+
+		if err = validations.ValidateVipDHCPAllocationWithIPv6(vipDhcpAllocation, string(cluster.MachineNetworks[index].Cidr)); err != nil {
 			return common.NewApiError(http.StatusBadRequest, err)
 		}
 	}
-	if params.ClusterUpdateParams.ServiceNetworkCidr != nil {
-		if err = network.VerifyClusterOrServiceCIDR(*params.ClusterUpdateParams.ServiceNetworkCidr); err != nil {
-			return common.NewApiError(http.StatusBadRequest, errors.Wrap(err, "Service network CIDR"))
+
+	if params.ClusterUpdateParams.ClusterNetworks != nil || params.ClusterUpdateParams.ServiceNetworks != nil ||
+		params.ClusterUpdateParams.MachineNetworks != nil {
+		// TODO: Support any number of subnets
```

Comment:
note to future us: once we support len(clusterNetworks) != len(blablaNetworks) we need to take proper care of this

Reply:
https://issues.redhat.com/browse/MGMT-7587